### ISSUE TITLE
libc++ recursive `setjmp()` macro causes `-Wdisabled-macro-expansion warning`

### ISSUE THREAD

**Issue by seanm** [llvm/llvm-project#27091]
|  |  |
| --- | --- |
| Bugzilla Link | [26717](https://llvm.org/bz26717) |
| Version | trunk |
| OS | MacOS X |

## Extended Description 
Consider this reduced/trivial C++ snippet:
```cpp
#include <setjmp.h>

int main()
{
  setjmp(0);
  return 0;
}
```

compiling with clang trunk gives:
```
/Users/sean/test.cxx:5:3: warning: disabled expansion of recursive macro [-Wdisabled-macro-expansion]
  setjmp(0);
  ^
/Users/sean/llvm/llvm-rel-install/bin/../include/c++/v1/setjmp.h:40:21: note: expanded from macro 'setjmp'
#define setjmp(env) setjmp(env)
                    ^
```
(If I use a `.c` file, I don't get the warning.)

As that's an llvm header (and has `#pragma GCC system_header`), I would hope not to get this warning.

**Comment by philnik777** [comment]
We don't have a `setjmp.h` anymore, so this should be fixed. I can't reproduce it either though. (OTOH it might be a macos-specific problem, so I wouldn't be able reproduce it either)

**Comment by seanm** [comment]
I just tried to repro with Xcode 16.2's clang, no longer repro. I guess it got fixed.
